### PR TITLE
[MM-54817] Convert ./components/admin_console/user_grid/user_grid_remove.tsx from Class Component to Function Component

### PR DESCRIPTION
**Pull request by sp6370**
#### Summary
This pull request migrates the existing user_grid_remove component to a react functional component. The component is primarily used in the Admin Console. 

One of the locations where the component is used: `admin_console/team_channel_settings/team/details/team_members/team_members.tsx`

#### Ticket Link
Github Issue: https://github.com/mattermost/mattermost/issues/24778
Jira: https://mattermost.atlassian.net/browse/MM-54817

#### Screenshots
1. Test passing for parent component (No specific test exist for the migrated component).
image

2. No specific UI changes. Adding component image here to make it easier for QA.
Screenshot from 2023-10-12 17-59-47

The specific component is the remove button on bottom right side of the image.

#### Release Note
```release-note
NONE
```

PS: Special thanks to @sinansonmez and @M-ZubairAhmed for helping me with my first issue.

### CODE REVIEW

**Review by larkox — CHANGES_REQUESTED** [review]
Thank you for your contribution! Just one small change. The second comment you can ignore.

**Review by sinansonmez — APPROVED** [review]
Apart from @larkox comments LGTM 🚀

File: webapp/channels/src/components/admin_console/user_grid/user_grid_remove.tsx [hunk 1]
```diff
@@ -12,31 +12,31 @@ type Props = {
     isDisabled?: boolean;
 }
 
-export default class UserGridRemove extends React.PureComponent<Props> {
-    private handleClick = (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+const UserGridRemove = (props: Props) => {
+    const {user, removeUser, isDisabled} = props;
+
+    const handleClick = (e: React.MouseEvent<HTMLAnchorElement>) => {
```
[the remark this answers is not in the document]
Comment (sp6370) [reply webapp/channels/src/components/admin_console/user_grid/user_grid_remove.tsx:21]: Got it! I am leaving it as it is!

File: webapp/channels/src/components/admin_console/user_grid/user_grid_remove.tsx [hunk 2]
```diff
@@ -12,31 +12,31 @@ type Props = {
     isDisabled?: boolean;
 }
 
-export default class UserGridRemove extends React.PureComponent<Props> {
-    private handleClick = (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+const UserGridRemove = (props: Props) => {
+    const {user, removeUser, isDisabled} = props;
```
[the remark this answers is not in the document]
Comment (sp6370) [reply webapp/channels/src/components/admin_console/user_grid/user_grid_remove.tsx]: I have addressed your suggestion. Please mark the conversation resolved, if it looks good.

**Review by larkox — APPROVED** [review]

**Review by sinansonmez — APPROVED** [review]
LGTM 🚀

**Comment by varghesejose2020** [comment]
@lindy65 I would like to test this PR as a QA.

**Comment by lindy65** [comment]
@varghesejose2020 - test server is ready

Log in to the test server using sysadmin credentials and go to the system console.  The component you're testing is explained below (from the contributor's summary at the top of this PR)

> 2. No specific UI changes. Adding component image here to make it easier for QA.
>    Screenshot from 2023-10-12 17-59-47
> 
> **The specific component is the remove button on bottom right side of the image.**

Let me know if you have any questions!

**Comment by varghesejose2020** [comment]
@lindy65 I tested the changes on the Test Server and it LGTM.
teamconf
Tested the Remove button on bottom right side .it working as expected

**Review by lindy65 — APPROVED** [review]
Thanks @sp6370 - looks good 👍 

Thanks for QA'ing @varghesejose2020 🎉

**Comment by larkox** [comment]
Thank you for the contribution!

**Pull request merged**